packages/frontend/src/redux/slices/sign/index.js

```diff
@@ -45,7 +46,8 @@ export const handleSignTransactions = createAsyncThunk(
                 const accountId = selectAccountId(getState());
 
                 try {
-                    transactionsHashes = await wallet.signAndSendTransactions(transactions, accountId);
+                    const signingWallet = shardInfo ? new Wallet(shardInfo) : wallet;
```

Comment:
If it's a private shard transaction a wallet pointing to the shard rpc it's used instead the default one.